### DISCUSSION TITLE
v4.10.0

### DISCUSSION THREAD

**Discussion by luu-alex** [ethereum/web3.js#7100] [Announcements]
#### web3

-   Now when existing packages are added in web3, will be avalible for plugins via context. (#7088)

#### web3-core

-   Now when existing packages are added in web3, will be avalible for plugins via context. (#7088)

#### web3-eth

-   `sendTransaction` in `rpc_method_wrappers` accepts optional param of `TransactionMiddleware` (#7088)
-   WebEth has `setTransactionMiddleware` and `getTransactionMiddleware` for automatically passing to `sentTransaction` (#7088)

#### web3-eth-ens

-   `getText` now supports first param Address
-   `getName` has optional second param checkInterfaceSupport

### web3-types

-   Added `result` as optional `never` and `error` as optional `never in type `JsonRpcNotification` (#7091)
-   Added `JsonRpcNotfication` as a union type in `JsonRpcResponse` (#7091)

### web3-rpc-providers

-   Alpha release 

#### web3-eth-ens

-   `getName` reverse resolution

## What's Changed
* release/v4.9.0 by @luu-alex in https://github.com/web3/web3.js/pull/7057
* fix(docs): intro by @danforbes in https://github.com/web3/web3.js/pull/7056
* coverage increase by @jdevcs in https://github.com/web3/web3.js/pull/7051
* 6992 getText update and getName fix by @jdevcs in https://github.com/web3/web3.js/pull/7047
* web3 external provider - Quicknode by @jdevcs in https://github.com/web3/web3.js/pull/7019
* fix(docs): quickstart by @danforbes in https://github.com/web3/web3.js/pull/7062
* Fix SNYK for 4.x by @avkos in https://github.com/web3/web3.js/pull/7064
* Use .snyk file for all projects by @avkos in https://github.com/web3/web3.js/pull/7071
* fix(docs): sidebar by @danforbes in https://github.com/web3/web3.js/pull/7063
* fix(docs): mastering providers by @danforbes in https://github.com/web3/web3.js/pull/7070
* Remove .snyk files by @avkos in https://github.com/web3/web3.js/pull/7073
* 100% Coverage web3-utills by @Muhammad-Altabba in https://github.com/web3/web3.js/pull/7042
* chore(docs): formatting by @danforbes in https://github.com/web3/web3.js/pull/7072
* update gas estimation test by @luu-alex in https://github.com/web3/web3.js/pull/7074
* fix(docs): contract-tutorial-hardhat-node by @danforbes in https://github.com/web3/web3.js/pull/7076
* fix(docs): header-nav by @danforbes in https://github.com/web3/web3.js/pull/7082
* fix(docs): introduction by @danforbes in https://github.com/web3/web3.js/pull/7083
* fix(docs): contracts-tutorial by @danforbes in https://github.com/web3/web3.js/pull/7084
* chore(docs): provider-tutorials by @danforbes in https://github.com/web3/web3.js/pull/7085
* fix(docs): mastering-accounts by @danforbes in https://github.com/web3/web3.js/pull/7089
* transaction middleware by @jdevcs in https://github.com/web3/web3.js/pull/7088
* add test coverage to web3-core by @luu-alex in https://github.com/web3/web3.js/pull/7091

**Full Changelog**: https://github.com/web3/web3.js/compare/v4.9.0...v4.10.0

<hr /><em>This discussion was created from the release <a href='https://github.com/web3/web3.js/releases/tag/v4.10.0'>v4.10.0</a>.</em>

**Comment by vittominacori** [comment]
Still not available on npm?
